lib/wadler/print.rb

```diff
@@ -75,17 +78,33 @@ def group(indent = 0, open_obj = '', close_obj = '',
     end
 
     # @param indent [Integer] nest indentation
+    # @param open_obj [String] nest opening delimiter
+    # @param close_obj [String] nest closing delimiter
     # @param break_type [Oppen::Token::BreakType] nest breaking type
     #
     # @return [Nil]
-    def nest(indent, break_type = Oppen::Token::BreakType::CONSISTENT, &block)
+    def nest(indent, open_obj = '', close_obj = '',
+             break_type = Oppen::Token::BreakType::CONSISTENT)
+      raise ArgumentError, "#{open_obj.nil? ? 'open_obj' : 'close_obj'} cannot be nil" \
+        if open_obj.nil? || close_obj.nil?
+
       @current_indent += indent
 
-      raise LocalJumpError if !block_given?
+      if !open_obj.empty?
+        text(open_obj)
+        self.break
+      end
+
+      begin
+        yield
+      ensure
+        @current_indent -= indent
+      end
+
+      return if close_obj.empty?
 
-      block.call
-    ensure
-      @current_indent -= indent
+      self.break
+      text(close_obj)
```

Comment:
I would move the whole "true" content in the begin/end.  And avoid return (but in error handling).

```ruby
    def nest(indent, open_obj = '', close_obj = '',
              break_type = Oppen::Token::BreakType::CONSISTENT)
       raise ArgumentError, "#{open_obj.nil? ? 'open_obj' : 'close_obj'} cannot be nil" \
         if open_obj.nil? || close_obj.nil?

       begin
         @current_indent += indent

         if !open_obj.empty?
           text(open_obj)
           self.break
         end

         yield

         if !close_obj.empty?
           self.break
           text(close_obj)
         end
       ensure
         @current_indent -= indent
       end
     end
```

Reply:
This doesn't work, we need to decrement current_indent before displaying the close_obj or else the we will not get the desired output.